ansible/roles/common/templates/wpa_supplicant.conf.j2

```diff
@@ -1,3 +1,14 @@
 ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev
 update_config=1
-country=KE
+country={{ wifi_country }}
+
+network={
+    ssid="{{ ssid1 }}"
+    psk="{{ psk_ssid1 }}"
+    priority=1
+}
+network={
+   ssid="{{ ssid2 }}"
+   psk="{{ psk_ssid2 }}"
+   priority=2
+}
```

Comment:
```suggestion
{% for network in wifi_settings.wifi_networks %}
network={
    ssid="{{ network.ssid }}"
    psk="{{ network.psk }}"
    priority={{ loop.index }}
}
```

Again this is not tested. And maybe instead of using `{{ loop.index }}`  you can specify the priority in the `vars.yaml` file.